python-mini-projects/projects/String_search_from_multiple_files/findstring.py

```diff
@@ -0,0 +1,33 @@
+import os
+
+text = input("input text : ")
+
+path = input("path : ")
+
+# os.chdir(path)
+
+
+def getfiles(path):
+    f = 0
+    os.chdir(path)
+    files = os.listdir()
+    # print(files)
+    for file_name in files:
+        abs_path = os.path.abspath(file_name)
+        if os.path.isdir(abs_path):
+            getfiles(abs_path)
+        if os.path.isfile(abs_path):
+            f = open(file_name, "r")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *f*. The resource is allocated by call *builtins.open*. Currently, there are execution paths that do not contain closure statements, for example, when an if condition at line 21 evaluates to true. To prevent this resource leak, close *f* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).